pkg/controllers/management/nodepool/nodepool.go

```diff
@@ -207,10 +216,29 @@ func (c *Controller) createOrCheckNodes(nodePool *v3.NodePool, simulate bool) (b
 		if v3.NodeConditionProvisioned.IsFalse(node) || v3.NodeConditionInitialized.IsFalse(node) || v3.NodeConditionConfigSaved.IsFalse(node) {
 			changed = true
 			if !simulate {
-				c.deleteNode(node, 2*time.Minute)
+				_ = c.deleteNode(node, 2*time.Minute)
+			}
+		}
+		// remove unreachable node with the unreachable taint & status of Ready being Unknown
+		q := getTaint(node.Spec.InternalNodeSpec.Taints, &unReachableTaint)
+		if q != nil && !simulate && nodePool.Spec.SelfHeal {
+			if isNodeReadyUnknown(node) {
+				start := q.TimeAdded.Time
+				if time.Since(start) > nodeNotReadyTimeout {
+					err = c.deleteNode(node, 0)
+					if err != nil {
+						return false, err
+					}
+				} else {
+					c.mutex.Lock()
+					if _, ok := c.syncmap[node.Name]; !ok {
+						c.syncmap[node.Name] = true
```

Comment:
since youre keeping track of all nodes maybe better to use the generated uid instead of the name here just in case you get two of the same at some point

Reply:
We are using the node.Name which is generated by Rancher and not set by the user. We use the node.Name in `NodeLister`

Reply:
Modified to use `node.Namespace + node.Name`